### ISSUE TITLE
It looks like `expect_lint: the_rule` also ignores the lint

### ISSUE THREAD

**Issue by dmrickey** [invertase/dart_custom_lint#160]
I pulled the repo and opened up the example to see if I could figure out why my local implementation wasn't working (still haven't figured out that part but that's a different question..). 

It looks like `expect_lint` also acts as ignore, the lint "underline" stops appearing, and doesn't show up in the problems aggregate window, if there is an expect comment

![image](https://github.com/invertase/dart_custom_lint/assets/3664822/6001886d-504c-476b-bfd3-ed0825be0a2c)

**Comment by rrousselGit** [comment]
Yes, that's the whole point of the feature. It is a way to write tests/examples without polluting the development environment. It would be a pain if the "problems" tabs in your IDE was polluted with hundreds of warnings when those warnings are in fact just tests for your lints.

TL;DR, if you _don't_ see a warning, that's because your lint correctly _did_ emit a warning at that line.   
If your lint did **not** emit a warning, you would see an error on the expect_lint instead. Which is what you see line 17 on your screenshot.
